Corona-Warn-App/src/test/java/de/rki/coronawarnapp/nearby/modules/diagnosiskeyprovider/DefaultDiagnosisKeyProviderTest.kt

```diff
@@ -62,135 +51,64 @@ class DefaultDiagnosisKeyProviderTest : BaseTest() {
-            )
+            googleENFClient.provideDiagnosisKeys(exampleKeyFiles)
+            googleENFClient.provideDiagnosisKeys(listOf(exampleKeyFiles[0]))
+            googleENFClient.provideDiagnosisKeys(listOf(exampleKeyFiles[1]))
             submissionQuota.consumeQuota(2)
         }
     }
 
     @Test
-    fun `normal key provision is used on newer ENF versions`() {
-        coEvery { googleAPIVersion.isAtLeast(GoogleAPIVersion.V16) } returns true
+    fun `key provision is used on newer ENF versions`() {
+        coEvery { googleAPIVersion.isAtLeast(GoogleAPIVersion.V15) } returns true
 
         val provider = createProvider()
 
-        runBlocking {
-            provider.provideDiagnosisKeys(exampleKeyFiles, exampleConfiguration, exampleToken)
-        }
+        runBlocking { provider.provideDiagnosisKeys(exampleKeyFiles) } shouldBe true
 
         coVerify(exactly = 1) {
-            googleENFClient.provideDiagnosisKeys(any(), any(), any())
-            googleENFClient.provideDiagnosisKeys(
-                exampleKeyFiles, exampleConfiguration, exampleToken
-            )
+            googleENFClient.provideDiagnosisKeys(any<List<File>>())
+            googleENFClient.provideDiagnosisKeys(exampleKeyFiles)
             submissionQuota.consumeQuota(1)
         }
     }
 
     @Test
-    fun `passing an a null configuration leads to constructing a fallback from defaults`() {
-        coEvery { googleAPIVersion.isAtLeast(GoogleAPIVersion.V16) } returns true
-
-        val provider = createProvider()
-        val fallback = ExposureConfiguration.ExposureConfigurationBuilder().build()
-
-        runBlocking {
-            provider.provideDiagnosisKeys(exampleKeyFiles, null, exampleToken)
-        }
-
-        coVerify(exactly = 1) {
-            googleENFClient.provideDiagnosisKeys(any(), any(), any())
-            googleENFClient.provideDiagnosisKeys(exampleKeyFiles, fallback, exampleToken)
-        }
-    }
-
-    @Test
-    fun `passing an a null configuration leads to constructing a fallback from defaults, legacy`() {
-        coEvery { googleAPIVersion.isAtLeast(GoogleAPIVersion.V16) } returns false
-
-        val provider = createProvider()
-        val fallback = ExposureConfiguration.ExposureConfigurationBuilder().build()
-
-        runBlocking {
-            provider.provideDiagnosisKeys(exampleKeyFiles, null, exampleToken)
-        }
-
-        coVerify(exactly = 1) {
-            googleENFClient.provideDiagnosisKeys(
-                listOf(exampleKeyFiles[0]), fallback, exampleToken
-            )
-            googleENFClient.provideDiagnosisKeys(
-                listOf(exampleKeyFiles[1]), fallback, exampleToken
-            )
-            submissionQuota.consumeQuota(2)
-        }
-    }
-
-    @Test
-    fun `quota is consumed silenently`() {
-        coEvery { googleAPIVersion.isAtLeast(GoogleAPIVersion.V16) } returns true
+    fun `quota is consumed silently`() {
```

Comment:
The quota is not consumed silently (passively, not interrupting), if the quota is exceeded, it returns `false` and we abort (in current code)?

Reply:
changed name to fun `provide diagnosis key when quota is empty`()